internal/services/redis/redis_cache_resource.go

```diff
@@ -820,12 +820,16 @@ func expandRedisConfiguration(d *pluginsdk.ResourceData) (*redis.RedisCommonProp
 	// nolint : staticcheck
 	v, valExists := d.GetOkExists("redis_configuration.0.rdb_backup_enabled")
 	if valExists {
-		if v := raw["rdb_backup_enabled"].(bool); v {
-			if connStr := raw["rdb_storage_connection_string"].(string); connStr == "" {
-				return nil, fmt.Errorf("The rdb_storage_connection_string property must be set when rdb_backup_enabled is true")
+		skuName := d.Get("sku_name").(string)
+		// rdb_backup_enabled is available when SKU is Premium
+		if strings.EqualFold(skuName, string(redis.SkuNamePremium)) {
```

Comment:
Can we also check if the SKU is not valid for this property, and return an error in case the user sets `rdb_backup_enabled = true`?

Reply:
Add a new constraint to cover the `rdb_backup_enabled = true` scenario.